### PR TITLE
#166115445-creates comments threads

### PR DESCRIPTION
**Pull request by shemaeric**
## What does this PR do?
 - Enables a user to create replies on a comment

## Description of Task to be completed?
User should be able to reply on a comment
## How should this be manually tested?
After you have logged in` and `created an article and commented on it.
- To test the reply on the comment make a POST request on 
 `localhost:3000/api/v1//comment/:commentId` with the JSON body of `reply` 
     You will get the JSON response with a `201` created.
 - To test the update reply on the comment make a PUT request on 
          `localhost:3000/api/v1/comment/:replyId/update` with an updated `reply` in the body
          you will get a `200` with a updated response.
  - To test the delete reply on the comment make a DELETE request on 
          `ocalhost:3000/api/v1/comment/:replyId/delete 
          you will get a `200` with a delete message response.
## Any background context you want to provide?
  - You have to log in first
  - Have a created article
  - Comment on an article
## What are the relevant pivotal tracker stories?
[User should be able to reply on comment 166115445](https://www.pivotaltracker.com/story/show/166115445)

### CODE REVIEW

**Comment by avpaul** [comment]
@aaronsekisambu  and @shemaeric It would be nice if you edit the PR description especially the **How should this be manually tested?** and make it more readable. The backticks(**`**) are making it weird!

File: controllers/comments.js [hunk 1]
```diff
@@ -192,25 +198,35 @@ export default {
...
       }
 
+      const findComment = await Comments.findOne({ where: { id: commentId } });
+      if (!findComment) {
+        return res.status(404).send({
+          status: res.statusCode,
+          error: 'The comment is not found'
+        });
+      }
+      const isAuthor = !!(id === CommentedArticle.author && role === 'admin');
```
Comment (avpaul) [inline controllers/comments.js]: Why not use or(**||**) here instead of using double negation?

File: controllers/comments.js [hunk 2]
```diff
@@ -192,25 +198,35 @@ export default {
...
+      if (!isAuthor && id !== findComment.author) {
```
Comment (avpaul) [inline controllers/comments.js]: I think here you also allowed the article author to be able to change comments? 🤔🤔I think that we were against this idea?

**Review by avpaul — CHANGES_REQUESTED** [review]
LGTM 👍 Except for where you allowed the article author to update a comment and a variable **status** that you didn't explain any where in the added code.

File: controllers/comments.js (hunk 2 above)
Comment (shemaeric) [reply controllers/comments.js]: Thank you @avpaul for pointing this out. Now it is good.

File: controllers/reply.js [hunk 3]
```diff
@@ -0,0 +1,166 @@
...
+
+      if (!findReply) {
+        return res.status(404).send({
+          status: res.statusCode,
+          error: 'The Reply on comment is not found'
+        });
+      }
+
+      if (req.user.id !== findReply.userId && req.user.role !== 'admin') {
+        return res.status(401).send({
```
Comment (niyobobo) [inline controllers/reply.js:87]: It looks like you allow admin to change someone's comment, I think it should be better if you restrict for only the author of the reply.

**Review by niyobobo — CHANGES_REQUESTED** [review]

File: controllers/reply.js (hunk 3 above)
Comment (shemaeric) [reply controllers/reply.js:87]: @niyobobo Thank you, the changes has been applied

**Review by avpaul — DISMISSED** [review]

**Review by fridolinho — DISMISSED** [review]

File: controllers/reply.js (hunk 3 above)
Comment (niyobobo) [reply controllers/reply.js:87]: @shemaeric not yet

**Review by niyobobo — DISMISSED** [review]

**Pull request merged**